src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -309,22 +309,32 @@ private async Task<Bundle> PerformContentResolverOperationAsync(ContentResolverO
         {
             ContentResolver resolver = GetContentResolver();
 
+            var contentResolverUri = GetContentProviderUriForOperation(Enum.GetName(typeof(ContentResolverOperation), operation));
+            _logger.Info($"Executing content resolver operation: {operation} URI: {contentResolverUri}");
+
             ICursor resultCursor = null;
-            await Task.Run(() => resultCursor = resolver.Query(AndroidUri.Parse(GetContentProviderUriForOperation(Enum.GetName(typeof(ContentResolverOperation), operation))),
+
+            await Task.Run(() => resultCursor = resolver.Query(AndroidUri.Parse(contentResolverUri),
                                                             !string.IsNullOrEmpty(_negotiatedBrokerProtocolKey) ? new string[] { _negotiatedBrokerProtocolKey } : null,
                                                             operationParameters,
                                                             null,
                                                             null)).ConfigureAwait(false);
 
             if (resultCursor == null)
             {
-                _logger.Error("[Android broker] An error occurred during the content provider operation.");
-                throw new MsalClientException(MsalError.CannotInvokeBroker, "[Android broker] Could not communicate with broker via content provider.");
+                _logger.Error($"[Android broker] An error occurred during the content provider operation {operation}.");
+                throw new MsalClientException(MsalError.CannotInvokeBroker, "[Android broker] Could not communicate with broker via content provider."
+                    + $"Operation: {operation} URI: {contentResolverUri}");
             }
 
             var resultBundle = resultCursor.Extras;
             resultCursor.Close();
```

Comment:
Please dispose the resultCursor instead of Close

Reply:
What is the reason for disposing instead of closing? Maybe we can dispose and close

Reply:
I looked at the content resolver provided by Xamarin to construct this method

https://github.com/xamarin/monodroid-samples/blob/0b301e8fd2da65ff442b5f1ed236c73ba3b963c2/PlatformFeatures/SimpleContentProvider/HomeScreen.cs#L53-L61

Reply:
It is a disposable object. Close would be part of dispose. Dispose is much cleaner that closes + frees memory for the local object.

Reply:
I will not hold the release for this. Close can be good enough. But dispose should be considered for later version.